### PR TITLE
Issue #167: Add options to specify base config and patch config

### PR DESCRIPTION
**Pull request by MEZk**
#167

### CODE REVIEW

File: checkstyle-tester/diff.groovy [hunk 1]
```diff
@@ -53,25 +61,36 @@ def getCliOptions(args) {
...
-    def patchBranch = cliOptions.patchBranch
-    def baseBranch = cliOptions.baseBranch
-    if (!isValidGitRepo(localGitRepo)
-           || !isExistingGitBranch(localGitRepo, patchBranch)
-           || baseBranch && !isExistingGitBranch(localGitRepo, baseBranch)) {
+    def baseConfig = cliOptions.baseConfig
+    def patchConfig = cliOptions.patchConfig
+    def config = cliOptions.config
+    if ((baseConfig || patchConfig || !config)
+            && (!baseConfig || !patchConfig || config)) {
```
Comment (rnveach) [inline checkstyle-tester/diff.groovy]: `if` statement looks confusing to read and understand to me.

File: checkstyle-tester/diff.groovy [hunk 2]
```diff
@@ -53,25 +61,36 @@ def getCliOptions(args) {
...
         valid = false
```
Comment (rnveach) [inline checkstyle-tester/diff.groovy]: Shouldn't we print what error is like we do in other `isValid`/`isExisting` methods below?

File: checkstyle-tester/diff.groovy (hunk 1 above)
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: @rnveach 
So did I when I created it. I have about 5 variants and all of them look confusing. The easiest one:
!(baseConfig && patchConfig && !config || !baseConfig && !patchConfig && config) . The condition that I used is a one of the minimal forms of the boolean function.

File: checkstyle-tester/diff.groovy (hunk 2 above)
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: @rnveach 
Maybe it is better to split theese boolean condition into 2 if branches and print error message.

Comment (MEZk) [inline checkstyle-tester/diff.groovy]: @rnveach 
Split into two if-blocks. Added error messages.

File: checkstyle-tester/diff.groovy (hunk 1 above)
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: @rnveach 
Split into two if-blocks.

File: checkstyle-tester/README.md [hunk 3]
```diff
@@ -104,7 +116,11 @@ The script receives the following set of command line arguments:
 
 **patchBranch** (p) - name of the branch with your changes;
 
-**checkstyleCfg** (c) - path to the file with Checkstyle configuration;
+**baseConfig** (bc) - path to the base checkstyle configuration file (required if patchConfig is specified);
+
+**patchConfig** (pc) - path to the patch checkstyle config file (required if baseConfig is specified);
+
+**config** (c) - path to the checkstyle config file (required if baseConfig and patchConfig are not secified);
```
Comment (MEZk) [inline checkstyle-tester/README.md]: @rnveach 
Done

File: checkstyle-tester/README.md [hunk 4]
```diff
@@ -104,7 +116,11 @@ The script receives the following set of command line arguments:
 
 **patchBranch** (p) - name of the branch with your changes;
 
-**checkstyleCfg** (c) - path to the file with Checkstyle configuration;
+**baseConfig** (bc) - path to the base checkstyle configuration file (required if patchConfig is specified);
+
+**patchConfig** (pc) - path to the patch checkstyle config file (required if baseConfig is specified);
```
Comment (MEZk) [inline checkstyle-tester/README.md]: @rnveach 
Done

File: checkstyle-tester/diff.groovy [hunk 5]
```diff
@@ -15,6 +15,14 @@ static void main(String[] args) {
             baseBranch = 'master'
         }
 
+        def baseConfig = cliOptions.baseConfig
+        def patchConfig = cliOptions.patchConfig
+        def config = cliOptions.config
+        if (config) {
```
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: @rnveach 
This looks weird but Groovy treats config as false if ```--config``` is not set and as true which may be cast to a string when ```--config``` is set)

File: checkstyle-tester/diff.groovy [hunk 6]
```diff
@@ -53,23 +61,57 @@ def getCliOptions(args) {
...
+        err.println "Error: $baseBranch is not an existing git branch!"
+        valid = false
+    }
+
+    return valid
+}
+
+def isValidCheckstyleConfigsCombination(config, baseConfig, patchConfig) {
+    def valid = true
+    if (config && patchConfig || config && baseConfig || config && patchConfig && baseConfig) {
```
Comment (romani) [inline checkstyle-tester/diff.groovy]: please make it like:
```
config && patchConfig 
    || config && baseConfig
    || config && patchConfig && baseConfig
```

File: checkstyle-tester/diff.groovy (hunk 5 above)
Comment (rnveach) [inline checkstyle-tester/diff.groovy]: Yes, I figured out what was going since I deal with JS alot.

**Comment by romani** [comment]
condition was rewritten in better way, I also failed to clearly read code before last changed. 
as my minor request for formatting is fixed , I am ok to merge.

File: checkstyle-tester/diff.groovy (hunk 6 above)
Comment (MEZk) [inline checkstyle-tester/diff.groovy]: @romani 
Done

**Pull request merged**